dissect/cstruct/expression.py

```diff
@@ -111,8 +112,8 @@ def tokenize(self) -> list[str]:
                 else:
                     pass
 
-                # number cannot end on x or b in the case of binary or hexadecimal notation
-                if token[-1] in ("x", "X", "b", "B"):
+                # number cannot end on x in the case of binary or hexadecimal notation
+                if token[-1] in ("x", "X"):
```

Comment:
This is to check that '0b' is not tokenized
```
if len(token) == 2 and  token[-1] in ("x", "X", "b", "B"):
```
This should fix it properly. I would commit it myself but I'm currently on holiday

Reply:
Awesome, thank you!